R/MixedModelsCommon.R

```diff
@@ -792,1293 +793,4580 @@
+      p <- .rstanPlotTrace(plotData[[i]])
+    } else if (options$samplingPlot == "stan_scat") {
+      p <- .rstanPlotScat(plotData[[i]])
+    } else if (options$samplingPlot == "stan_hist") {
+      p <- .rstanPlotHist(plotData[[i]])
+    } else if (options$samplingPlot == "stan_dens") {
+      p <- .rstanPlotDens(plotData[[i]])
+    } else if (options$samplingPlot == "stan_ac") {
+      p <- .rstanPlotAcor(plotData[[i]])
+    }
+
+
+    if (options$samplingPlot %in% c("stan_hist", "stan_dens")) {
+      p <- jaspGraphs::themeJasp(p, sides = "b")
+      p <- p + ggplot2::theme(
+        axis.title.y = ggplot2::element_blank(),
+        axis.text.y  = ggplot2::element_blank(),
+        axis.ticks.y = ggplot2::element_blank()
+      )
+      p <- p + ggplot2::labs(x = varName)
+    } else{
+      p <- jaspGraphs::themeJasp(p)
+    }
+    if (options$samplingPlot == "stan_trace") {
+      p <- p + ggplot2::theme(plot.margin = ggplot2::margin(r = 10 * (nchar(options$iteration - options$warmup) - 2)))
+    }
+    plots$plotObject <- p
+
+    diagnosticPlots[[names(plotData)[i]]] <- plots
+  }
+
+}
+
+# helper functions
+.mmVariableNames  <- function(varName, variables) {
+  for (vn in variables) {
+    inf <- regexpr(vn, varName, fixed = TRUE)
+    if (inf[1] != -1) {
+      varName <- paste0(
+        substr(varName, 0, inf[1] - 1),
+        substr(varName, inf[1], inf[1] + attr(inf, "match.length") - 1),
+        " (",
+        substr(
+          varName,
+          inf[1] + attr(inf, "match.length"),
+          nchar(varName)
+        )
+      )
+    }
+  }
+  varName <- gsub(":", paste0(")", jaspBase::interactionSymbol), varName, fixed = TRUE)
+  varName <- paste0(varName, ")")
+  varName <- gsub(" ()", "", varName, fixed = TRUE)
+  return(varName)
+}
+.mmAddCoefNameStanova <- function(samples, par, coefs_name){
+  # this is a mess but the stanova::stanova_samples returns an incomplete variable names
+
+  coefs_trend <- attr(samples, "estimate")
+  coefs_trend <- gsub("trend ('", "", coefs_trend, fixed = TRUE)
+  coefs_trend <- gsub("')", "", coefs_trend, fixed = TRUE)
+  coefs_trend <- strsplit(coefs_trend, ",")
+
+  for(cft in coefs_trend){
+    if(cft %in% strsplit(par, ":")[[1]] && !grepl(.unv(cft), coefs_name)){
+      coefs_name <- paste0(coefs_name, jaspBase::interactionSymbol, .unv(cft))
+    }
+  }
+
+  return(coefs_name)
+
+}
+.mmGetPlotSamples <- function(model, pars, options) {
+  matrix_diff <-
+    stanova::stanova_samples(model,
+                             return = "array",
+                             diff_intercept = options$show == "deviation")
+
+  if (length(pars) == 1) {
+    samples <- matrix_diff[[pars]]
+    coefs   <- dim(matrix_diff[[pars]])[2]
+
+    plotData <- list()
+
+    for (cf in 1:coefs) {
+
+      coefs_name <-
+        paste(.unv(unlist(
+          strsplit(dimnames(samples)$Parameter[cf], ",")
+        )), collapse = ":")
+      coefs_name <- gsub(" ", "", coefs_name, fixed = TRUE)
```

Comment:
if this is removing leading and/ or trailing whitespace, then `trimws()` is more appropriate (and faster).

Reply:
it's not removing trailing spaces but spaces inside of the characters